lib/purl_fetcher/client/reader.rb

```diff
@@ -18,11 +19,17 @@ def collection_members(druid)
     end
   end
 
+  # @return [Array<Hash<String,String>>] a list of hashes where the key is a digest and the value is a filepath/filename
+  def files_by_digest(druid)
+    get("/purls/druid:#{druid.delete_prefix('druid:')}", {})
+      .fetch("files_by_md5", [])
+  end
```

Comment:
Not sure if this existing class is the best place to put this method. I assumed that the name of the class implied it was for read-only operations (GETs), since the collections-by-druid operation looks like the only GET in the client. FWIW, I was unable to find any consumers of `PurlFetcher::Client::Reader` anywhere in the `sul-dlss` GH org, so this may also be dead code. 🤷🏻‍♂️